src/components/swap/ConfirmSwapModal.tsx

```diff
@@ -80,48 +81,66 @@ function useConfirmModalState({
       steps.push(ConfirmModalState.PERMITTING)
     }
     steps.push(ConfirmModalState.PENDING_CONFIRMATION)
-    setPendingModalSteps(steps)
+    return steps
   }, [allowance])
 
   const { chainId } = useWeb3React()
   const trace = useTrace()
   const maximumAmountIn = useMaxAmountIn(trade, allowedSlippage)
 
-  const startSwapFlow = useCallback(async () => {
-    setApprovalError(undefined)
-    if (allowance.state === AllowanceState.REQUIRED) {
-      // Starts the approval process, by triggering either the Token Approval or the Permit signature.
-      try {
-        if (allowance.needsSetupApproval) {
+  const catchUserReject = async (e: any, errorType: PendingModalError) => {
```

Comment:
does there have to be a separate `PendingModalError` type based on step - can you just use the same `ConfirmModalState` variable and deduce based on that

Reply:
yes, but on the other hand these two components don't need to be so tightly coupled. maybe we want to use the error UI in a different modal in the future ?